core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -373,4 +434,196 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String getWriteableName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryShardContext context) throws IOException {
+        MultiMatchQuery multiMatchQuery = new MultiMatchQuery(context);
+        if (analyzer != null) {
+            if (context.analysisService().analyzer(analyzer) == null) {
+                throw new QueryShardException(context, "[" + NAME + "] analyzer [" + analyzer + "] not found");
+            }
+            multiMatchQuery.setAnalyzer(analyzer);
+        }
+        if (slop != null) {
```

Comment:
Some of these checks can potentially be removed after changing internal field to primitve types / using defaults.